client/src/components/revealComplete.js

```diff
@@ -21,47 +26,72 @@ const ButtonContainer = styled.div`
 const getRevealedWaifus = async (WETStore) => {
   const oldWaifuIds = WETStore.ownedItems.map(item => item.id);
   await WETStore.syncOwnedItems();
+  const newWaifuIds = WETStore.ownedItems.map(item => item.id);
   const newWaifus = WETStore.ownedItems;
   return newWaifus.filter(waifu => !oldWaifuIds.includes(waifu.id));
+  return newWaifuIds.filter(newId => !oldWaifuIds.includes(newId));
 }
 
 const RevealComplete = observer(({ show }) => {
   const rootStore = useContext(RootStoreContext);
   const { WETStore } = rootStore;
-  const [revealedWaifus, setRevealedWaifus] = useState([])
+  const [revealedWaifuIds, setRevealedWaifuIds] = useState([]);
+  const [loading, setLoading] = useState(false);
   
-  useEffect(() => {
-    if (show) {
-      getRevealedWaifus(WETStore).then((value) => {
-        setRevealedWaifus(value);
-      })
-    }
-  })
+  useEffect(
+    () => {
+      if (show) {
+        setLoading(true);
+        getRevealedWaifus(WETStore)
+        .then((value) => {
+          setRevealedWaifuIds(value);
+          setLoading(false);
+        }).catch((err) => {
+          setLoading(false);
+        });
+      }
+    },
+    [show, WETStore]
+  )
 
   return (
     <StyledRevealComplete>
       <Popup
         show={show}
         content={
           <BoxContent>
-            <Header>Waifu Revealed!!!</Header>
             {
-              show && (
-                revealedWaifus.map((item) => {
-                  return <WaifuCard cardIndex={item.id} />
-                })
-              )
+              loading 
+                ? 
+                  <Loader
+                    className="wallet-waifu-loader"
+                    color="#FBE55F"
+                    size="80px"
+                  />
+                :
+              <>
+                <Header>{revealedWaifuIds.length} Waifu{revealedWaifuIds.length > 1 && 's'} Revealed!!!</Header>
+                <RevealWaifusContainer>
+                  {
+                    show && (
+                      revealedWaifuIds.map((id) => {
+                        return <WaifuCard cardIndex={id} key={id} />
```

Comment:
The `id` is the Waifu object, not the string id - this causes the image to load incorrectly & be linked incorrectly because JS stringifies it as [object Object] in the img/href URL. Other than that, seems to work. :)

Reply:
Fixed: https://github.com/chase-manning/waifusion-dungeon/pull/63/commits/2d736059896b5a3433311fc5cb720c0aba8a1205